### ISSUE TITLE
[BUG] Scraper doesn't let code execution finish before taking screenshot

### ISSUE THREAD

**Issue by alexrockhill** [pyvista/pyvista#5277]
### Describe the bug, what's wrong, and what you expected.

The `mne.viz.Brain` inherits from `PyVistaFigure` and is scraped in `sphinx-example-gallery` examples but `PyVistaFigure.set_camera` calls don't take effect before the image is snapped.

### Steps to reproduce the bug.

Here is an example artifact, the brain plots at the end are way zoomed in and they should have the camera zoomed out after the `show_view` which calls `set_camera` call: https://output.circle-artifacts.com/output/job/7614b75f-9bc0-4d8e-a59b-983dfdbead6d/artifacts/0/dev/auto_examples/ieeg_locate.html

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Mon Dec 04 10:00:48 2023 PST

                OS : Linux
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 31.2 GiB
       Environment : IPython
       File system : ext4
        GPU Vendor : Intel
      GPU Renderer : Mesa Intel(R) Xe Graphics (TGL GT2)
       GPU Version : 4.6 (Core Profile) Mesa 23.2.1-1pop0~1699535617~22.04~6cd9a72~dev
  MathText Support : False

  Python 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:08:06)
  [GCC 11.3.0]

           pyvista : 0.41.1
               vtk : 9.2.6
             numpy : 1.23.5
        matplotlib : 3.7.2
            scooby : 0.7.1
             pooch : v1.7.0
           imageio : 2.27.0
         pyvistaqt : 0.11.0
             PyQt5 : 5.15.7
           IPython : 8.14.0
        ipyvtklink : 0.2.2
        ipywidgets : 8.0.6
             scipy : 1.10.1
              tqdm : 4.65.0
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

**Comment by banesullivan** [comment]
I see you're using `pyvistaqt` here... perhaps the threading involved with that is the problem? Otherwise this may be tough for us to track down... Would you be able to try to create some sort of minimal, reproducible example? I suspect in that process you may tease out a usage issue with pyvistaqt, but otherwise its going to be tough for us to dive into the MNE code to track this down

**Comment by alexrockhill** [comment]
Hmmm, I think it's just the `pyvista` scraper. I'll try adding a call to update the `plotter` here https://github.com/pyvista/pyvista/blob/9273ed67f6fa721cfb75492f3f0ef541009227c6/pyvista/plotting/utilities/sphinx_gallery.py#L84-L86 and see if that fixes it.

**Comment by banesullivan** [comment]
Re-opening since that PR is being reverted

**Comment by tkoyama010** [comment]
resolved by #5321 .
